dashboard/views.py

```diff
@@ -652,39 +687,44 @@ def are_weighted_assignments_hidden(course_id, df):
         else:
             logger.info(f"few weighted assignments in course {course_id} are hidden")
             return True
+    return False
 
+class BinningGrade(NamedTuple):
+    value: int
+    bin_index: int
+    binning_all: bool
 
-def find_binning_grade_value(grades):
+def find_binning_grade_value(grades: list) -> BinningGrade:
     fifth_item = grades[4]
     next_to_fifth_item = grades[5]
     if next_to_fifth_item - fifth_item > 2:
-        BinningGrade = get_binning_grade()
-        return BinningGrade(value=fifth_item, index=4, binning_all=False)
+        return BinningGrade(value=fifth_item, bin_index=4, binning_all=False)
     else:
         return binning_logic(grades, fifth_item)
 
 
-def is_odd(num):
+def is_odd(num: int) -> bool:
     if num % 2 == 0:
         return False
     else:
         return True
 
 
-def show_dashed_line(grade, BinningGrade):
+def show_dashed_line(grade: int, binning_grade: BinningGrade) -> bool:
     """
     logic determine to show dashed line or not.
     :param grade:
     :param BinningGrade:
     :return:
     """
-    if BinningGrade.binning_all or grade > 96 or grade < 2:
+    if binning_grade.binning_all or grade > 96 or grade < 2:
         return False
     else:
         return True
 
 
-def check_if_grade_qualifies_for_binning(grade, fifthElement):
+def check_if_grade_qualifies_for_binning(grade: int,
+                                         fifthElement: str) -> bool:
     # case: 96.7, 94.76,
     if int(grade) - int(fifthElement) > 1:
```

Comment:
Based on this line, `grade` value that is passed in is actually  `str`, the annotations need to be changed

Reply:
Hmm, okay. I was updating some of these now that we have type ints to be the correct values. I can make them both str's or probably at least Union[str, float]. int doesn't seem appropriate though.

Reply:
So the logic `int(grade) - int(fifthElement) > 1` is still appropriate, maybe there is a better way of doing it but the idea is correct

Reply:
This is where type checking is handy. ;)
````
def find_binning_grade_value(grades: List[str]) -> BinningGrade:
    fifth_item = grades[4]
    next_to_fifth_item = grades[5]
    if next_to_fifth_item - fifth_item > 2:
```

With this logic grades can't be a str, because `next_to_fifth_item - fifth_item` isn't possible with str. I'm guessing grades is a float it looks like it is in the model.

It originally comes from
`    grades = df['current_grade'].values.tolist()